mindsdb/interfaces/datastore/datastore.py

```diff
@@ -62,6 +59,7 @@ def get_datasources(self, name=None):
                 datasource_arr.append(datasource)
             except Exception as e:
                 log.error(e)
+                self.delete_datasource(datasource_record.name)
```

Comment:
This we shouldn't do, as in the current process a "broken" datasource could be "under construction".
Also, there should be no reason why a datasource could be "Broken" so for now let's keep them in the database to see what can go wrong to cause them to be so.